### PR TITLE
Various changes in sysbox-in-docker setup

### PR DESCRIPTION
**Pull request by rodnymolina**
Refer to individual commits for details.

### CODE REVIEW

File: sysbox-in-docker/sysbox [hunk 1]
```diff
@@ -144,42 +144,26 @@ function kernel_config_setup() {
...
-  local l0_egress_mtu
-  local l1_egress_mtu
-
   # Identify default egress iface.
   local egress_iface=$(ip route show | awk '/default via/ {print $5}')
   if [ -z "${egress_iface}" ]; then
 	  return
   fi
 
   # Obtain mtu value associated to the test-container's egress interface.
```
Comment (ctalledo) [inline sysbox-in-docker/sysbox:160]: nit: "test-container" -> "privileged-container" ?

File: sysbox-in-docker/sysbox [hunk 2]
```diff
@@ -259,17 +243,32 @@ function maxs_setup() {
   sysctl -w kernel.keys.maxbytes=$((20*20000)) &> /dev/null
 }
 
+# Sysbox performs certain operations that require read-write access to sysfs. In
+# the scenario where this script executes (i.e. privileged containers), sysfs is
+# usually mounted in read-write mode, but there is one bug (apparently in
```
Comment (ctalledo) [inline sysbox-in-docker/sysbox:248]: I suspect the problem here is that when launching a privileged ** pod **, the pause container's spec has sysfs mounted read-only, so a similar mount shows up in the sibling container (somehow). I doubt the problem is a bug in containerd itself.  In other words, when launching a privileged pod, the sysfs mount is showing up as read-only instead of read-write.

File: sysbox-in-docker/Makefile [hunk 3]
```diff
@@ -34,7 +34,7 @@ help: ## Show supported docker images
 LAUNCH_INSTRUCTION := docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker \
                 -v /var/tmp/sysbox-var-lib-docker:/var/lib/docker \
                 -v /var/tmp/sysbox-var-lib-sysbox:/var/lib/sysbox \
-                -v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro \
+                -v /lib/modules/$$(uname -r):/lib/modules/$$(uname -r):ro \
```
Comment (ctalledo) [inline sysbox-in-docker/Makefile:37]: Why was this necessary (out of curiosity)?

**Review by ctalledo — APPROVED** [review]

File: sysbox-in-docker/sysbox (hunk 1 above)
Comment (rodnymolina) [reply sysbox-in-docker/sysbox:160]: Missed that one, thanks.

File: sysbox-in-docker/sysbox (hunk 2 above)
Comment (rodnymolina) [reply sysbox-in-docker/sysbox:248]: Could be, makes sense.

File: sysbox-in-docker/Makefile (hunk 3 above)
Comment (rodnymolina) [reply sysbox-in-docker/Makefile:37]: I guess you are referring to the double "$". See that we are echo'ying this instruction to stdout, so a "$$" is required to avoid variable interpolation and force command execution instead.
